cardano-cli/src/Cardano/CLI/Common/Parsers.hs

```diff
@@ -29,43 +30,48 @@ import qualified Text.Parsec.Language as Parsec
 import qualified Text.Parsec.String as Parsec
 import qualified Text.Parsec.Token as Parsec
 
-pCardanoEra :: Parser AnyCardanoEra
-pCardanoEra = asum
-  [ Opt.flag' (AnyCardanoEra ByronEra) $ mconcat
-    [ Opt.long "byron-era"
-    , Opt.help "Specify the Byron era"
-    ]
-  , Opt.flag' (AnyCardanoEra ShelleyEra) $ mconcat
-    [ Opt.long "shelley-era"
-    , Opt.help "Specify the Shelley era"
-    ]
-  , Opt.flag' (AnyCardanoEra AllegraEra) $ mconcat
-    [ Opt.long "allegra-era"
-    , Opt.help "Specify the Allegra era"
-    ]
-  , Opt.flag' (AnyCardanoEra MaryEra) $ mconcat
-    [ Opt.long "mary-era"
-    , Opt.help "Specify the Mary era"
-    ]
-  , Opt.flag' (AnyCardanoEra AlonzoEra) $ mconcat
-    [ Opt.long "alonzo-era"
-    , Opt.help "Specify the Alonzo era"
-    ]
-  , Opt.flag' (AnyCardanoEra BabbageEra) $ mconcat
-    [ Opt.long "babbage-era"
-    , Opt.help "Specify the Babbage era (default)"
-    ]
-  , Opt.flag' (AnyCardanoEra ConwayEra) $ mconcat
-    [ Opt.long "conway-era"
-    , Opt.help "Specify the Conway era"
-    ]
+pCardanoEra :: EnvCli -> Parser AnyCardanoEra
+pCardanoEra envCli =
+  asum $ mconcat
+    [ [ Opt.flag' (AnyCardanoEra ByronEra) $ mconcat
+        [ Opt.long "byron-era"
+        , Opt.help "Specify the Byron era"
+        ]
+      , Opt.flag' (AnyCardanoEra ShelleyEra) $ mconcat
+        [ Opt.long "shelley-era"
+        , Opt.help "Specify the Shelley era"
+        ]
+      , Opt.flag' (AnyCardanoEra AllegraEra) $ mconcat
+        [ Opt.long "allegra-era"
+        , Opt.help "Specify the Allegra era"
+        ]
+      , Opt.flag' (AnyCardanoEra MaryEra) $ mconcat
+        [ Opt.long "mary-era"
+        , Opt.help "Specify the Mary era"
+        ]
+      , Opt.flag' (AnyCardanoEra AlonzoEra) $ mconcat
+        [ Opt.long "alonzo-era"
+        , Opt.help "Specify the Alonzo era"
+        ]
+      , Opt.flag' (AnyCardanoEra BabbageEra) $ mconcat
+        [ Opt.long "babbage-era"
+        , Opt.help "Specify the Babbage era (default)"
+        ]
+      , Opt.flag' (AnyCardanoEra ConwayEra) $ mconcat
+        [ Opt.long "conway-era"
+        , Opt.help "Specify the Conway era"
+        ]
 
-  -- NEW-ERA-ADD-NEW: When a new era is added, add a new flag here.
-  -- NEW-ERA-SET-DEFAULT: When a new era is working, select a new default above and below.
+        -- NEW-ERA-ADD-NEW: When a new era is added, add a new flag here.
+        -- NEW-ERA-SET-DEFAULT: When a new era is working, select a new default above and below.
+      ]
 
-    -- Default for now:
-  , pure (AnyCardanoEra BabbageEra)
-  ]
+    , maybeToList $ pure <$> envCliAnyCardanoEra envCli
+
+      -- TODO is this default needed anymore?
```

Comment:
So what would be the behaviour if this default gets removed? Will specifying era on the CLI or environment variable become mandatory?

Reply:
I think perhaps the default will be initialised at a higher level and passed in via `EnvCli`.